File: app/src/main/java/app/revanced/bilibili/settings/fragments/BiliRoamingSettingsFragment.java
package app.revanced.bilibili.settings.fragments;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.preference.Preference;

import app.revanced.bilibili.utils.KtUtils;
import app.revanced.bilibili.utils.Utils;

public class BiliRoamingSettingsFragment extends BiliRoamingBaseSettingFragment {
    public BiliRoamingSettingsFragment() {
        super("biliroaming_settings");
    }

    @Override
    public void onCreatePreferences(@Nullable Bundle savedInstanceState, @Nullable String rootKey) {
        super.onCreatePreferences(savedInstanceState, rootKey);
        Preference preference = findPreference("debug");
        if (preference != null)
            preference.setSummary(Utils.getString("biliroaming_debug_summary", KtUtils.getLogFile().getAbsolutePath()));
    }
}

File: app/src/main/java/app/revanced/bilibili/settings/fragments/BiliRoamingSettingsFragment.kt
package app.revanced.bilibili.settings.fragments

import android.os.Bundle
import androidx.preference.Preference
import app.revanced.bilibili.utils.Utils
import app.revanced.bilibili.utils.onClick

class BiliRoamingSettingsFragment : BiliRoamingBaseSettingFragment("biliroaming_settings") {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        super.onCreatePreferences(savedInstanceState, rootKey)
        findPreference<Preference>("reboot")?.onClick { Utils.reboot(); true }
    }
}
